cpp/unreal_plugins/UrdfBot/Source/UrdfBot/UrdfJointComponent.cpp

```diff
@@ -48,56 +52,124 @@ void UUrdfJointComponent::initializeComponent(UrdfJointDesc* joint_desc, UUrdfLi
+        float damping = FMath::RadiansToDegrees(joint_desc->damping_) * m_to_cm;
+        float force_limit = FMath::RadiansToDegrees(joint_desc->effort_) * m_to_cm;
 
-            // use twist degree for single angular freedom with best PhysX optimization
-            ConstraintInstance.SetAngularTwistLimit(EAngularConstraintMotion::ACM_Free, 0.0f);
-            ConstraintInstance.SetOrientationDriveTwistAndSwing(true, false);
-            ConstraintInstance.SetAngularVelocityDriveTwistAndSwing(true, false);
-            ConstraintInstance.SetAngularDriveParams(spring, damping, force_limit);
-            break;
+        // use twist degree for single angular freedom with best PhysX optimization
+        ConstraintInstance.SetAngularTwistLimit(EAngularConstraintMotion::ACM_Free, 0.0f);
+        ConstraintInstance.SetOrientationDriveTwistAndSwing(true, false);
+        ConstraintInstance.SetAngularVelocityDriveTwistAndSwing(true, false);
+        ConstraintInstance.SetAngularDriveParams(spring, damping, force_limit);
+        break;
+    }
+    case UrdfJointType::Prismatic: {
+        float linear_limit_size = (joint_desc->upper_ - joint_desc->lower_) * m_to_cm;
+        float spring = joint_desc->spring_ * m_to_cm * m_to_cm;
+        float damping = joint_desc->damping_ * m_to_cm * m_to_cm;
+        float force_limit = joint_desc->effort_ * m_to_cm * m_to_cm;
+
+        ConstraintInstance.SetLinearXLimit(ELinearConstraintMotion::LCM_Limited, linear_limit_size);
+        ConstraintInstance.SetLinearDriveParams(spring, damping, force_limit);
+        ConstraintInstance.SetLinearPositionDrive(true, false, false);
+        break;
+    }
+    case UrdfJointType::Fixed:
+        break;
+    case UrdfJointType::Floating: {
+        ConstraintInstance.SetLinearXLimit(ELinearConstraintMotion::LCM_Free, 0.0f);
+        ConstraintInstance.SetLinearYLimit(ELinearConstraintMotion::LCM_Free, 0.0f);
+        ConstraintInstance.SetLinearZLimit(ELinearConstraintMotion::LCM_Free, 0.0f);
+        ConstraintInstance.SetAngularTwistLimit(EAngularConstraintMotion::ACM_Free, 0.0f);
+        ConstraintInstance.SetAngularSwing1Limit(EAngularConstraintMotion::ACM_Free, 0.0f);
+        ConstraintInstance.SetAngularSwing2Limit(EAngularConstraintMotion::ACM_Free, 0.0f);
+        break;
+    }
+    case UrdfJointType::Planar: {
+        float linear_limit_size = (joint_desc->upper_ - joint_desc->lower_) * m_to_cm;
+
+        ConstraintInstance.SetLinearXLimit(ELinearConstraintMotion::LCM_Limited, linear_limit_size);
+        ConstraintInstance.SetLinearYLimit(ELinearConstraintMotion::LCM_Limited, linear_limit_size);
+        break;
+    }
+    default:
+        ASSERT(false);
+        break;
+    }
+
+    SetDisableCollision(true);
+    SetConstrainedComponents(parent_link_component, NAME_None, child_link_component, NAME_None);
+}
+
+float UUrdfJointComponent::getAction() {
+
+    switch (control_type_) {
```

Comment:
be consistent with your return and break statements within these nested switch statements, case blocks should be more indented than the containing switch keyword.

Reply:
fixed